pkgs/build-support/rust/import-cargo-lock.nix

```
@@ -304,16 +304,24 @@ let
             done
 
             for crate in ${toString depCrates}; do
-              # Link the crate directory, removing the output path hash from the destination.
-              ln -s "$crate" $out/$(basename "$crate" | cut -c 34-)
-
               if [ -e "$crate/.cargo-config" ]; then
                 key=$(sed 's/\[source\."\(.*\)"\]/\1/; t; d' < "$crate/.cargo-config")
                 if [[ -z ''${keysSeen[$key]} ]]; then
                   keysSeen[$key]=1
                   cat "$crate/.cargo-config" >> $out/.cargo/config.toml
                 fi
               fi
+              # In monorepos, it's possible to have multiple crates with the
+              # same name, pointing at potentially different versions. As such,
+              # link the only top definition, but collect config from all (done
+              # above).
+              outCrate=$out/$(basename "$crate" | cut -c 34-)
+              if [ -e $outCrate ]; then
+                echo "Warning: $out/$(basename "$crate" | cut -c 34-) already exists, not overwriting it"
```

This is just the value of `$outCrate`:

```suggestion
                echo "Warning: $outCrate already exists, not overwriting it"
```